CMakeLists.txt

```diff
@@ -46,6 +49,28 @@ add_subdirectory(demos)
 add_subdirectory(benchs)
 add_subdirectory(tutorial/cpp)
 
+
+# Use jemalloc for memory allocation
+if (FAISS_ENABLE_JEMALLOC)
+  # Find jemalloc
+  if (FAISS_DEFAULT_JEMALLOC_LOCATION)
+    # If FAISS_DEFAULT_JEMALLOC_LOCATION is set, find_package will look for jemalloc in default location
+    # specified in cmake/FindJemalloc.cmake
+    find_package(jemalloc REQUIRED)
+    # If FAISS_DEFAULT_JEMALLOC_LOCATION is not set, (in case faiss is a sub-project of another project)
+    # the values to following cmake variables must be set by the parent project:
+    # JEMALLOC_INCLUDE_DIR -> where to find jemalloc/jemalloc.h
+    # JEMALLOC_LIBRARY_RELEASE -> where to find jemalloc library
+    # Check if the parent project has set the required variables and if not, throw an error
+  endif()
+  if (DEFINED JEMALLOC_LIBRARY_RELEASE AND DEFINED JEMALLOC_INCLUDE_DIR AND EXISTS "${JEMALLOC_LIBRARY_RELEASE}" AND EXISTS "${JEMALLOC_INCLUDE_DIR}")
```

Comment:
At a bare minimum, this variable should be called JEMALLOC_LIBRARIES, not JEMALLOC_LIBRARY_RELEASE since it could easily be a debug library rather than a release library.

Reply:
done